docs/cpp/c-cpp-properties-schema-reference.md

```diff
@@ -13,37 +13,106 @@ This article explains the scheme for the c_cpp_properties.json settings file.
 
 Looking to get started with configuring your project? See [Configure Intellisense](/docs/cpp/configure-intellisense.md).For more information about changing these settings, see [Customizing Default Settings](/docs/cpp/customize-default-settings-cpp.md).
 
-## Example
+## Example of Variables
+
+Note, this is an example of all fields. You do not need to specify all fields in your `c_cpp_properties.json` file. The extension will automatically fill in any missing fields with default values.
 
 ```json
 {
-    "env" : {
-        "myDefaultIncludePath": [
-            "${workspaceFolder}",
-            "${workspaceFolder}/include"
+    "env": {
+        "myIncludePath": [
+            "${workspaceFolder}/include",
+            "${workspaceFolder}/src"
         ],
-        "myCompilerPath": "/usr/local/bin/gcc-7"
+        "myDefines": [
+            "DEBUG",
+            "MY_FEATURE=1"
+        ]
     },
     "configurations": [
+        {
+            "name": "Linux",
+            "compilerPath": "/usr/bin/gcc",
+            "compilerArgs": [
+                "-m32"
+            ],
+            "intelliSenseMode": "linux-gcc-x86",
+            "includePath": [
+                "${myIncludePath}",
+                "/usr/include"
```

Comment:
I know you warn people not to copy this, but we don't want to hint that it's ok to add `/usr/include` in this file under any circumstances since it will likely be added by the compiler.

Reply:
Make up some other path, like `${workspaceFolder}/3rdparty/include` or something like that